### PR TITLE
Gtd helpers tests

### PR DESCRIPTION
**Pull request by KaiSpencer**
Some tests for the public helper functions for gtd.
Slight readability refactor on the tasks callback

First PR, new to lua, be nice 😄 

- [x] stylua

### CODE REVIEW

File: lua/neorg/modules/core/gtd/helpers/module.lua [hunk 1]
```diff
@@ -65,14 +65,25 @@ module.public = {
...
             data.type,
             ["task"] = function()
-                return (
-                        (
-                            type(data.contexts) == "table" and not vim.tbl_isempty(data.contexts)
-                            or (type(data["waiting.for"]) == "table" and not vim.tbl_isempty(data["waiting.for"]))
-                        ) and not data.inbox
-                    )
-                    or (type(data["time.due"]) == "table" and not vim.tbl_isempty(data["time.due"]))
-                    or (type(data["time.start"]) == "table" and not vim.tbl_isempty(data["time.start"]))
```
Comment (danymat) [inline lua/neorg/modules/core/gtd/helpers/module.lua:75]: You shouldn't change this code, if you're only doing test suites. I'm not even sure this is correct though

**Comment by danymat** [comment]
Very nice, the test suites are okay to me, I'm just not found of re-xriting the `is_processed` function here.

**Comment by KaiSpencer** [comment]
Hey @danymat, Ive refactored the tests to more nicely match the changes you made to the is_proccessed function.

**Review by danymat — APPROVED** [review]

**Comment by danymat** [comment]
Merged, thanks for your help !

**Pull request merged**